crates/swc_ecma_minifier/tests/projects/output/angular-1.2.5.js

```diff
@@ -2757,7 +2756,7 @@
                     $watchCollection: function(obj, listener) {
                         var oldValue, newValue, self = this, changeDetected = 0, objGetter = $parse(obj), internalArray = [], internalObject = {}, oldLength = 0;
                         return this.$watch(function() {
-                            if (newValue = objGetter(self), isObject(newValue)) {
```

Comment:
Same here. We need to prove that `objGetter(self)` does not modify `isObject`.

Reply:
Same here. terser would inline newValue as long ass isObject is declared.

Reply:
No, the declaration is not the condition. I changed the definition of `isObject` to `var isObject = globalThis.isObject;` and terser preserves 

```js
newValue = objGetter(self);
                        var newLength, key;
                        if (!isObject(newValue)) {
                            if (oldValue !== newValue) {
                                oldValue = newValue;
                                changeDetected++
                            }
```

https://gist.github.com/kdy1/c4247944514115421aaed0c9288b6d38#file-beutified-js-L5291-L5293

https://gist.github.com/kdy1/c4247944514115421aaed0c9288b6d38